### ISSUE TITLE
Use adaptive time step

### ISSUE THREAD

**Issue by Amirrgrbn** [sfepy/sfepy#822]
Hello, 
I want to have a force, using increments with time steps; it would be the cyclic load so loading and unloading, I wan the unlodqing part takes place much slower also symmetric (starts and ends at 0 value for force). 
I have already seent the Adaptive example in ball problem, but it wasnt helpful; would you pleased let me know how can i possibily do it?

**Comment by vlukes** [comment]
Hi @Amirrgrbn,
You can use `parametric_hook` to control your time step. Have a look at e.g. [diffusion/poisson_parametric_study.py](https://sfepy.org/doc-devel/examples/diffusion-poisson_parametric_study.html), where the hook is employed.

**Comment by rc** [comment]
Or maybe the other way round. If the force is given by a material function, you could make it so that the unloading is slower easily even with a fixed time-step. Are you using only pseudo-time-stepping (quasistatic problem) to gradually change the force, or is it really a time-dependent problem?

**Comment by Amirrgrbn** [comment]
Thanks for both suggestions, yes it is only pseudo time step, and the force is defined by material function. I was thinking of changing the time step rather than changing the force increment.

**Comment by rc** [comment]
Note that changing the increment is definitely easier.

**Comment by rc** [comment]
Yes, you would need to integrate stress * n over the surface with the normal n, see [here](https://en.wikipedia.org/wiki/Stress_(mechanics)#General_stress). Essentially, `dw_surface_ltr` (or `de_surface_ltr`) should be usable (use v = 1).

**Comment by rc** [comment]
Try attaching a code + mesh demonstrating the problem.
